vector.js

```diff
@@ -173,10 +173,11 @@
     value = value || +0.0;
     type = type || Float64Array;
     var data = new type(count),
+        isValueFn = typeof value === 'function',
         i;
 
     for (i = 0; i < count; i++)
-      data[i] = value;
+      data[i] = (isValueFn && value()) || value;
```

Comment:
If `value()` is a function returning `0` the condition will become false and it will erroneously pass the function itself as a value. This results in a `Vector` filled with `NaN` fields.

Test:

```javascript
Vector.fill(5, function () {
  return 0;
});
// [NaN, NaN, NaN, NaN, NaN]
```

I also think the index `i` should be passed to the value function.